draft-gpew-priv-ppm.md

```diff
@@ -820,28 +832,28 @@ been collected. This is the case if the aggregator is the leader and has
 completed a collect request for a batch containing the report (see
 {{collect-flow}}), or if the aggregator is a helper and has completed an
 aggregate-share request for a batch containing the report (see
-{{aggregate-share-request}}). It also checks if the report has been observed.
+{{aggregate-share-request}}). It also checks if the report has been aggregated.
 This is the case if the report was used in a previous aggregate request (see
 {{aggregate-init-request}} and {{aggregate-request}}).
 
-* If the report has not yet been observed but is contained by a batch that has
-  been collected, then the aggregator fails with error `batch-collected`. This
-  prevents additional reports from being aggregated after its batch has already
-  been collected.
+* If the report has never been aggregated but is contained by a batch that has
```

Comment:
I'm not sure I agree with the change of wording from "observed" to "aggregated" here.

There will be a gap of time from when a client upload completes to when it is first included in an aggregation job; my interpretation would be that those reports were "observed" (or perhaps "received") but not yet "aggregated". I think we want this error logic to cover reports that have not yet been included in an aggregation job, right?

Reply:
That makes sense; my only note is that the client's Upload Request section (4.2.2) refers to this section as part of a `SHOULD`, so I suppose this logic may also apply to client upload endpoints. I think this is OK since the relevant bullet point (this one) is not meaningfully changed by the wording change. This would matter for the `report-replayed` point, but that's not mentioned by the client upload section for now; we might need to deal with it later but it's fine for now.

Reply:
Hmm, interesting. We should to try to work this out now, if possible. I think what we want to say that, before a batch is collected, two reports with the same nonce MUST NOT be aggregated twice. Furthermore, the leader SHOULD report to the client if it attempted to upload a report with a previously observed nonce.